plugins/modules/activation.py

```diff
@@ -261,15 +280,43 @@ def create_params(
         activation_params["k8s_service_name"] = module.params["k8s_service_name"]
 
     # Get the webhook ids
-    webhooks_ids = None
-    if not is_aap_24 and module.params.get("webhooks"):
-        webhooks_ids = []
-        for item in module.params["webhooks"]:
-            webhook_id = lookup_resource_id(module, controller, "webhooks", item)
-            if webhook_id is not None:
-                webhooks_ids.append(webhook_id)
-    if webhooks_ids is not None:
-        activation_params["webhooks"] = webhooks_ids
+    sources = []
+    if not is_aap_24 and module.params.get("event_streams"):
+        activation_params["event_streams"] = []
+        try:
+            sources = controller.get_one_or_many(
+                f"rulebooks/{rulebook_id}/sources", name=module.params["rulebook_name"]
+            )
+        except EDAError as e:
+            module.fail_json(msg=f"Failed to get rulebook source list: {e}")
+
+        for elem in module.params.get("event_streams"):
+            event = elem.copy()
+            if event.get("source_index") and event.get("source_name"):
+                module.fail_json(
+                    msg="source_index and source_name options are mutually exclusive."
+                )
+
+            if event.get("source_index") is None and event.get("source_name") is None:
+                module.fail_json(
+                    msg="You must specify one of the options source_index or source_name."
+                )
+
+            if event.get("source_index"):
+                try:
+                    event["source_name"] = sources[event["source_index"]]
+                except IndexError as e:
+                    module.fail_json(
+                        msg=f"The specified source_index {event['source_index']} is out of range: {e}"
+                    )
+                event.pop("source_index")
+            elif event.get("source_name") and not any(
+                d["name"] == event.get("source_name") for d in sources
+            ):
+                module.fail_json(
+                    msg=f"The specified source_name {event.get('source_name')} does not exist."
+                )
+            activation_params["event_streams"].append(event)
```

Comment:
The param for the api is not "event_streams" but "source_mappings". 
This is to align the name with the UI. 
The payload for source_mapping is a yaml string that must meet the following schema:
```
  - source_name: 
    event_stream_id: 
    event_stream_name:
    rulebook_hash:
```

So, as it is now, what we have to do is: fetch as well the event_stream_id, and also the rulebook_hash for each event_stream and convert the list in a YAML to be passed to the "source_mappings" param.

Reply:
I tried to apply those suggestions. Should I also rename the module's option to source_mappings or event_streams is ok?

Reply:
I think event_streams is ok because is the name in the UI. Nobody else has raised concerns about it. This is a mismatch between the UI and the API, so the module is in the middle. I don't have a strong opinion about but I think in the doubt, pairing it with the UI makes more sense. Do you agree?

Reply:
Makes sense.